pkg/plugins/policies/meshratelimit/plugin/xds/configurer.go

```diff
@@ -26,11 +26,15 @@ func RateLimitConfigurationFromPolicy(rl *api.LocalHTTP) *rate_limit.RateLimitCo
 	var status uint32
 	if rl.OnRateLimit != nil {
 		for _, h := range pointer.Deref(rl.OnRateLimit.Headers) {
-			headers = append(headers, &rate_limit.Headers{
+			header := &rate_limit.Headers{
 				Key:    h.Key,
 				Value:  h.Value,
-				Append: pointer.Deref(h.Append),
-			})
+				Append: true,
+			}
+			if h.Append != nil {
```

Comment:
At least with k8s [when you use `default`, there's always a value](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting). Does our schema machinery also handle setting defaults so we don't need to do a nil check here?

Reply:
I think we'd need to check this.

Reply:
checking

Reply:
Btw isn't this k8s behavior going to be a problem for merging?!

Reply:
Our universal doesn't set the value if it's absent and has default I've just checked.

Reply:
So anyway in that specific case that doesn't cause any problem. But I believe that can be problematic in other places